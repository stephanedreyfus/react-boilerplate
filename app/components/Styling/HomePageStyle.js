/** Styled components for HomePage. */

import styled from 'styled-components';

const mainColor = 'mediumvioletred';

export const Wrapper = styled.section`
  padding: 4em;
  background: lavender;
`;

export const Greeting = styled.h1`
  font-size: 1.5em;
  text-align: center;
  color: ${mainColor};
  background: #e0e2e5;
  border-radius: 6px;
  border: 0.1em solid ${mainColor};
  box-shadow: 5px 6px 8px ${mainColor};
  margin: 3em auto auto auto;
  width: 50%;
`;

export const Link = styled.a`
  font-size: 1em;
  text-align: center;
  color: ${mainColor};

  ${Link}:hover & {
    color: black;
    transition: color 1s;
  }
`;
