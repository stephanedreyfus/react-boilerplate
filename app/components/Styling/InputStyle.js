/** Stylng for InputForm component. */
import styled from 'styled-components';

export const Form = styled.form`
  width: 85%;
  margin: auto;
`;

export const Label = styled.label`
  font-family: 'Roboto', sans-serif;
`;

export const Input = styled.input`
  width: 84%;
  padding: 0.2em;
  margin: 0.4em;
  border-radius: 10px;
  border-width: 1px;
`;

export const PhraseButton = styled.button`
  font-family: 'Roboto', sans-serif;
`;
