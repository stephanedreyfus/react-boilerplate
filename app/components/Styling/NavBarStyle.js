/* Styling for NavLinks in NavBar.js component. */
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

export const NavWrapper = styled.div`
  margin: 0;
  font-family: 'Montserrat', Helvetica, sans-serif;
`;

export const NavBar = styled.div`
  overflow: hidden;
  background-color: #333;
`;

export const LinkElem = styled(NavLink)`
  float: left;
  display: block;
  color: mediumvioletred;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  font-size: 17px;

  &:hover {
    background-color: #ddd;
    color: mediumvioletred;
  }

  &.active {
    color: lightseagreen;
  }
`;
