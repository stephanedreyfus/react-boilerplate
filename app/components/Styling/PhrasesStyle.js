/** Styled components for Phrases */

import styled from 'styled-components';

const mainColor = 'mediumvioletred';

export const Wrapper = styled.section`
  background: #e0e2e5;
  height: 20em;
  width: 42%;
  border-radius: 6px;
  border: 0.1em solid ${mainColor};
  box-shadow: 5px 6px 8px ${mainColor};
  margin: 3em auto auto auto;
`;

export const DisplayTitle = styled.div`
  font-family: 'Montserrat', sans-serif;
  color: ${mainColor};
  text-align: center;
  border-bottom: 0.1em solid ${mainColor};
  margin: 1em 3em;
`;

export const Display = styled.div`
  font-family: 'Roboto', sans-serif;
  width: 85%;
  padding: 1em;
  margin: 0.5em 1em;
  border-radius: 10px;
  border-width: 1px;
`;

export const ListContaine = styled.ul`
  list-style: none;
`;

export const PhraseList = styled.li`
  margin: auto;
`;
