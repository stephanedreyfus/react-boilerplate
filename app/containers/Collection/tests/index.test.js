import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { Collection } from '../index';
import { DisplayTitle } from '../../../components/Styling/PhrasesStyle';
import configureStore from '../../../configureStore';

describe('<Collection />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it.skip('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(<Collection dispatch={dispatch} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <Collection />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should render its heading', () => {
    const {
      container: { firstChild },
    } = render(<DisplayTitle>Loading Phrases, Please Wait</DisplayTitle>);
    expect(firstChild).toMatchSnapshot();
  });
});
