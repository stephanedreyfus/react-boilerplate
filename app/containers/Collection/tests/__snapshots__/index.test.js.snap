// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Collection /> Should render and match the snapshot 1`] = `
<section
  class="PhrasesStyle__Wrapper-sc-13tp1yz-0 sADTq"
>
  <div
    class="PhrasesStyle__DisplayTitle-sc-13tp1yz-1 gxwBMv"
  >
    Collected Phrases
  </div>
  <ul
    class="PhrasesStyle__PhraseList-sc-13tp1yz-3 dfJTiS"
  >
    <li
      class="PhrasesStyle__ListItem-sc-13tp1yz-4 hWFumD"
    />
  </ul>
</section>
`;

exports[`<Collection /> should render its heading 1`] = `
<div
  class="PhrasesStyle__DisplayTitle-sc-13tp1yz-1 gxwBMv"
>
  Loading Phrases, Please Wait
</div>
`;
